test/controllers/EditSetControllerSpec.scala

```diff
@@ -244,72 +291,107 @@ class EditSetControllerSpec extends BaseSpec {
         "endDateMonth"              -> "10",
         "endDateYear"               -> "2020",
         "legalStatus"               -> "ref.1",
+        "placeOfDeposit"            -> "2",
         "action"                    -> "save"
       )
 
     "when the action is to save the record" when {
+
+      val validValuesForSaving = validValues ++ Map("action" -> "save")
+
       "fail" when {
         "and yet preserve the CCR" when {
           "there are errors" in {
-            val ccrToAssert = "COAL 80/80/1"
             val blankScopeAndContentToFailValidation = ""
-            val request = CSRFTokenHelper.addCSRFToken(
-              FakeRequest(POST, "/edit-set/1/record/COAL.2022.V5RJW.P/edit")
-                .withFormUrlEncodedBody(
-                  "ccr"                       -> ccrToAssert,
-                  "oci"                       -> "1234",
-                  "scopeAndContent"           -> blankScopeAndContentToFailValidation,
-                  "formerReferenceDepartment" -> "1234",
-                  "coveringDates"             -> "1234",
-                  "startDate"                 -> "1234",
-                  "endDate"                   -> "1234",
-                  "action"                    -> "save"
-                )
-                .withSession(SessionKeys.token -> validSessionToken)
-            )
-            val editRecordPage = route(app, request).get
+            val values = validValuesForSaving ++ Map("scopeAndContent" -> blankScopeAndContentToFailValidation)
+
+            val result = submitWhileLoggedIn(1, "COAL.2022.V5RJW.P", values)
 
-            status(editRecordPage) mustBe BAD_REQUEST
-            val document = asDocument(editRecordPage)
-            document must haveHeading(s"TNA reference: $ccrToAssert")
+            status(result) mustBe BAD_REQUEST
+            assertPageAsExpected(
+              asDocument(result),
+              ExpectedPage(
```

Comment:
Do you think it's worth having a default expected page case class instance, with valid values, at the top level of this spec? Then tests expecting invalid values could `.copy(....` the properties that are being asserted for that spec.

The bonuses that I see would be:
* cut down on noise in tests
* make it explicit what kind of error message / value is being asserted.

Reply:
I prefer tests which are explicit to all expected values; so I don't see it as noise.